### ISSUE TITLE
Add an option to redirect to "canonical" URL

### ISSUE THREAD

**Issue by quicksketch** [backdrop/backdrop-issues#868]
This issue is a bit of a followup to https://github.com/backdrop/backdrop-issues/issues/865, where we identified that URLs in Backdrop can be accessed a number of places. Say the alias was set to "contact" for node 1:
- /contact
- /?q=contact
- index.php/?q=contact
- /node/1
- ?q=node/1
- index.php/?q=node/1

We should have an option to akin to the "Global Redirect" module in core to make it so that all these different URLs are not accessible. In #865, the "Clean URLs" page may be renamed to "URL handling". After adding the option for "Use HTTPS for canonical URLs" we could add another checkbox below it for:

`[ ] Redirect to canonical URL`

As the option states, if a user ended up on a non-aliased, non-clean, old alias, or non-HTTP(S) path, they would be redirected to the single canonical URL. This single solution could provide a lightweight alternative to both global redirect and redirect modules.

**Comment by klonos** [comment]
:+1:

**Comment by docwilmot** [comment]
Great that this is moving along. I Googled what canonical means, and I kinda get that its a good idea. I would ask though that we find a simpler easy to grasp description to go with this option, or change the label itself from `[ ] Redirect to canonical URL` to something simple. I'd give suggestions but I've forgotten again what exactly it means.

**Comment by Graham-72** [comment]
LOL here!

**Comment by quicksketch** [comment]
How about:

```
[ ] Redirect to clean or aliased URLs if available
```

After thinking about this functionality and the possible trouble you could get into with it, I don't think we should try to handle HTTP/HTTPS redirects, or any redirects involving domain name changes (e.g. www/non-www or old-domain.com to new-domain.com). The potential for redirecting to broken URLs or loops conflicting with .htaccess redirects seems too high. Redirecting to clean or aliased versions only seems a much safer and more common issue. So in a way, renaming this option to avoid the term "canonical" (which would also deal with domain name and protocol) makes more sense to me anyway.

**Comment by quicksketch** [comment]
It doesn't look like we'll get this in for 1.4.0, though we did add Redirect module (which covers creating manual redirects from one path to another) in https://github.com/backdrop/backdrop-issues/issues/905.

In that issue, we discussed separating out the global redirect functionality into somewhere in the menu system, or keeping it in the Redirect module. It looks like generally there's little overlap between manual redirects and global redirects, so we could go either way with that yet.
